server/controllers/chartsDataController/index.js

```diff
@@ -21,21 +22,34 @@ const show = async (req, res, next) => {
       parsedQueryParams.filters,
       userSites
     )
+    const filters = filtersService.filters
     const chart = await appDb.collection(collections.charts).findOne({
       _id: new ObjectId(chart_id),
     })
-    const chartService = new BarChartService(appDb, chart)
-    const filters = filtersService.filters
-    const participants = await ParticipantsModel.allForAssessment(
+    const chartService = new BarChartService(chart)
+    const chartProcessor = new BarChartDataProcessor(
+      chart,
+      chartService.generateStudyTargetTotals(filters.sites)
+    )
+    const dataStream = await ParticipantsModel.allForAssessment(
       appDb,
       chart,
       filtersService
     )
+    dataStream.on('data', (doc) => chartProcessor.processDocument(doc))
```

Comment:
can we keep all this logic in the service so we're not polluting the controller with it?

Reply:
[403d065](https://github.com/AMP-SCZ/dpdash/pull/631/commits/403d065498c9a44a8c0963abfe236bc601a2615c)
👍